tool/src/nanobind/formatter.rs

```diff
@@ -0,0 +1,201 @@
+//! This module contains functions for formatting types
+
+use crate::c::{CFormatter, CAPI_NAMESPACE};
+use diplomat_core::hir::{self, StringEncoding, TypeContext, TypeId};
+use std::borrow::Cow;
+
+/// This type mediates all formatting
+///
+/// All identifiers from the HIR should go through here before being formatted
+/// into the output: This makes it easy to handle reserved words or add rename support
+///
+/// If you find yourself needing an identifier formatted in a context not yet available here, please add a new method
+///
+/// This type may be used by other backends attempting to figure out the names
+/// of C types and methods.
+pub(crate) struct PyFormatter<'tcx> {
```

Comment:
issue: I think this type should be containing the cpp formatter instead of reimplementing it

tool code should just call `formatter.cpp.whatever()`